src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -21,10 +21,7 @@ def branch_package(workflow_filters = {})
     add_branch_request_file(package: target_package)
 
     # SCMs don't support statuses for tags, so we don't need to report back in this case
-    unless scm_webhook.tag_push_event?
-      create_or_update_subscriptions(target_package, workflow_filters)
-      report_to_scm(workflow_filters)
-    end
+    create_or_update_subscriptions(target_package, workflow_filters) unless scm_webhook.tag_push_event?
```

Comment:
We should evaluate the workflow_filters, in the workflow level, not inside the steps. As it is now, when I create a new PR via github UI (git generates a push event), or push to github in a new branch, even if i have an event filter as described here https://github.com/openSUSE/open-build-service/wiki/Better-SCM-CI-Integration#event-filter, filtering just for pull requests, I still run all the workflow steps.

Reply:
We're passing the workflow filters to this method in order to have them in the event subscription payload. This is needed for the `ReportToScmJob`:

https://github.com/openSUSE/open-build-service/blob/944d7ec129ffe0aa2fb1fd05ced5d035d88c9d0a/src/api/app/jobs/report_to_scm_job.rb#L34-L40

Reply:
As for running all steps when we pass an event filter, this sounds definitely wrong what we're doing. Steps shouldn't be called if the webhook event doesn't match the event filter.

Reply:
I had a look and it should be working fine. See this [comment](https://github.com/openSUSE/open-build-service/pull/12154#issuecomment-1028856756). Is it not working somehow?